test/compilable/dtoh_enum.d

```diff
@@ -86,6 +89,42 @@ END_ENUM_TYPE(const char*, EnumWithImplicitType, ENUMWITHIMPLICITTYPE, enumwithi
 ENUM_CONSTANT_NUMERIC(int32_t, AnonMixedOne, 1)
 ENUM_CONSTANT_NUMERIC(int64_t, AnonMixedTwo, 2LL)
 ENUM_CONSTANT(const char*, AnonMixedA, "a")
+
+
+BEGIN_ENUM(STC, STC, stc)
+    ENUM_KEY(int32_t, a, 1, STC, STC, stc, STC)
+    ENUM_KEY(int32_t, b, 2, STC, STC, stc, STC)
+END_ENUM(STC, STC, stc)
+
+ENUM_CONSTANT(STC, STC_D, (STC)3)
+
+// ignoring non-cpp struct Foo because of linkage
+BEGIN_ENUM_TYPE(Foo, MyEnum, MYENUM, myenum)
```

Comment:
This doesn't make much sense, but I added it to check that there's at least no segfault.